core/modules.rs

```diff
@@ -20,10 +19,31 @@ use std::rc::Rc;
 use std::task::Context;
 use std::task::Poll;
 
-pub type SourceCodeInfoFuture =
-  dyn Future<Output = Result<SourceCodeInfo, ErrBox>>;
+/// EsModule source code that will be loaded into V8.
+///
+/// Users can implement `Into<ModuleInfo>` for different file types that
+/// can be transpiled to valid EsModule.
+///
+/// Found module URL might be different from specified URL
+/// used for loading due to redirections (like HTTP 303).
+/// Eg. Both "https://example.com/a.ts" and
+/// "https://example.com/b.ts" may point to "https://example.com/c.ts"
+/// By keeping track of specified and found URL we can alias modules and avoid
+/// recompiling the same code 3 times.
```

Comment:
Why 3?

Reply:
Don't know; old comment, PTAL at my TODO comment below

Reply:
Having some bookkeeping of redirects sounds useful.

Reply:
CC @piscisaureus